### PR TITLE
Enhancement @types/three Added definition for THREE.FilmPass  #14042

### PR DESCRIPTION
**Pull request by apurvaojas**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/mrdoob/three.js/blob/dev/examples/js/shaders/FilmShader.js>> <<https://github.com/mrdoob/three.js/blob/dev/examples/js/postprocessing/FilmPass.js>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by dhritzkiv** [comment]
Great.

Can this class definition be made to extend `THREE.Pass` (this may need to be added)? There are a lot of properties and methods that are inherited, and it would be a shame to lose those in the definitions.

**Comment by apurvaojas** [comment]
Hi Hritzkiv,

As i checked there is no class named `THREE.Pass` in the definitions. Also other Shader passes doesn't extends `THREE.Pass`  For example  [MaskPass](https://github.com/mrdoob/three.js/blob/dev/examples/js/postprocessing/MaskPass.js). I have added the definitions for the variables and functions in FilmPass. 

Although It work without it, as SMAAPASS definitions are working. Please Find below the links for SMAAPASS definitions and source for reference.
 
<<https://github.com/DefinitelyTyped/DefinitelyTyped/blob/8829824eae983d97d2be9b22d2cae05336ee8936/types/three/three-smaapass.d.ts>>

<<https://github.com/mrdoob/three.js/blob/dev/examples/js/postprocessing/SMAAPass.js>>

And if the definitions are suitable then merge it.

Thanks

**Comment by dhritzkiv** [comment]
hi @apurvaojas.

Currently, THREE.Pass doesn't exist in the definitions, but I was hoping that you could quickly add it, and extend FilmPass off of it. The source for it is here: https://github.com/mrdoob/three.js/blob/a885290f59ff6b1692b5cccd8e7ab5b2bba33946/examples/js/postprocessing/EffectComposer.js#L163-L189

This will help keep definitions for future effects based off of THREE.Pass cleaner and easier to maintain.

**Comment by apurvaojas** [comment]
Hi @dhritzkiv,

I have added the definition for THREE.Pass and extend it in THREE.FilmPass.

File: types/three/three-filmpass.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+export class FilmPass extends Pass {
+    constructor(noiseIntensity: number, scanlinesIntensity: number, scanlinesCount: number, grayscale: boolean);
+    
+    scene: Scene;
+    camera: Camera;
+    uniforms: IUniform;
+    material: Material;
+    quad: Mesh;
+
+   render(renderer: WebGLRenderer, writeBuffer: WebGLRenderTarget, readBuffer: WebGLRenderTarget, delta: number, maskActive?: boolean): void;
```
Comment (dhritzkiv) [inline types/three/three-filmpass.d.ts]: You can likely remove this render method declaration, as it's declared on `Pass`

**Review by dhritzkiv — CHANGES_REQUESTED** [review]

**Comment by dhritzkiv** [comment]
Other than that, it's good to go

File: types/three/three-filmpass.d.ts (hunk 1 above)
Comment (apurvaojas) [reply types/three/three-filmpass.d.ts]: removed :thumbsup:

**Review by dhritzkiv — APPROVED** [review]

**Pull request merged**